src/main/scala/it/pagopa/interop/authorizationprocess/api/impl/ClientApiServiceImpl.scala

```diff
@@ -127,26 +127,10 @@ final case class ClientApiServiceImpl(
     val operationLabel: String = s"Retrieving client $clientId"
     logger.info(operationLabel)
 
-    def isConsumerOrProducer(organizationId: UUID, client: PersistentClient)(implicit
-      ec: ExecutionContext
-    ): Future[Boolean] = {
-      def isProducer(): Future[Boolean] = Future
-        .traverse(client.purposes.map(_.eService.eServiceId))(catalogManagementService.getEServiceById)
-        .map(eservices => eservices.map(_.producerId).contains(organizationId))
-
-      if (client.consumerId == organizationId) Future.successful(true) else isProducer()
-
-    }
-
     val result: Future[Client] = for {
-      requesterUuid        <- getOrganizationIdFutureUUID(contexts)
-      clientUuid           <- clientId.toFutureUUID
-      client               <- authorizationManagementService.getClient(clientUuid)
-      isConsumerOrProducer <- isConsumerOrProducer(requesterUuid, client)
-      _                    <- Future
-        .failed(OrganizationNotAllowedOnClient(client.id.toString, requesterUuid))
-        .unlessA(isConsumerOrProducer)
-
+      requesterUuid <- getOrganizationIdFutureUUID(contexts)
+      clientUuid    <- clientId.toFutureUUID
+      client        <- authorizationManagementService.getClient(clientUuid)
```

Comment:
as per task description, relationships should be hidden if the requester is not the consumer